pkg/opensearch/opensearch.go

```diff
@@ -80,39 +80,35 @@ func (ds *OpenSearchDatasource) QueryData(ctx context.Context, req *backend.Quer
 // the associated Stats query. It then adds these parameters to the originating query so
 // the Stats query can be created later.
 func handleServiceMapPrefetch(ctx context.Context, osClient client.Client, req *backend.QueryDataRequest) error {
-	var serviceMapQuery backend.DataQuery
-	var serviceMapQueryIndex int
 	for i, query := range req.Queries {
-		model, _ := simplejson.NewJson(query.JSON)
+		model, err := simplejson.NewJson(query.JSON)
+		if err != nil {
+			return err
+		}
 		queryType := model.Get("queryType").MustString()
 		luceneQueryType := model.Get("luceneQueryType").MustString()
 		serviceMapRequested := model.Get("serviceMap").MustBool(false)
 		if queryType == Lucene && luceneQueryType == "Traces" && serviceMapRequested {
```

Comment:
style nit: totally optional, but you could change this to
```suggestion
		if queryType != Lucene || luceneQueryType != "Traces" || !serviceMapRequested {
```
and a continue (to make the code more left aligned)